ios/Classes/SwiftChargebeeFlutterSdkPlugin.swift

```diff
@@ -214,12 +214,31 @@ extension SKProduct {
     func toMap() -> [String: Any?] {
         let map: [String: Any?] = [
             "productId": productIdentifier,
-            "productPrice": price.description,
+            "productPrice": price.doubleValue,
             "productTitle": localizedTitle,
-            "currencyCode": priceLocale.currencyCode
+            "currencyCode": priceLocale.currencyCode,
+            "subscriptionPeriod": subscriptionPeriod()
         ]
         return map
     }
+    func subscriptionPeriod() -> [String:Any?]?  {
+        var subscriptionPeriod: [String: Any?]? = nil;
+        let period:String = {
+            switch self.subscriptionPeriod?.unit {
+            case .day: return "day"
+            case .week: return "week"
+            case .month: return "month"
+            case .year: return "year"
+            case .none: return ""
+            case .some(_): return ""
+            }
+        }()
```

Comment:
Replace with a function/method call. Also is it possible to just return the rawValue of the enum?

Reply:
Done

Reply:
Can you refactor this to be a method call?